### PR TITLE
fixes `SearchAnchor` leak [prod-leak-fix]

### PR DESCRIPTION
**Pull request by Dimilkalathiya**
This PR contributes to #141198

- Fixes memory leak on `SearchAnchor`
- Adds opt-in test

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

**Comment by Dimilkalathiya** [comment]
cc: @polina-c

 Linux docs_test seems to be failing i saw logs it seems to be for generating docs? i don't have idea about this perhaps re-running test would fix this? let me know if it has something to do with PR changes.

**Comment by polina-c** [comment]
Yes, let's start with just restarting.

**Comment by Dimilkalathiya** [comment]
And now it seems like a bunch of are failing as-well

Can't find anything useful on the log

**Comment by polina-c** [comment]
Flutter-tree went red recently: https://flutter-dashboard.appspot.com/#/build
That means tests were failing at master.
I updated branch again.

**Comment by Dimilkalathiya** [comment]
Looks like it's fixed ✨👀

File: packages/flutter/lib/src/material/search_anchor.dart [hunk 1]
```diff
@@ -620,23 +629,25 @@ class _SearchViewRoute extends PopupRoute<_SearchViewRoute> {
       child: AnimatedBuilder(
         animation: animation,
         builder: (BuildContext context, Widget? child) {
-          final Animation<double> curvedAnimation = CurvedAnimation(
+           curvedAnimation ??= CurvedAnimation(
```
Comment (polina-c) [inline packages/flutter/lib/src/material/search_anchor.dart:632]: Remove extra space in front of `curvedAnimation`

File: packages/flutter/lib/src/material/search_anchor.dart [hunk 2]
```diff
@@ -620,23 +629,25 @@ class _SearchViewRoute extends PopupRoute<_SearchViewRoute> {
...
-          final Rect viewRect = _rectTween.evaluate(curvedAnimation)!;
+          final Rect viewRect = _rectTween.evaluate(curvedAnimation!)!;
           final double topPadding = showFullScreenView
-            ? lerpDouble(0.0, MediaQuery.paddingOf(context).top, curvedAnimation.value)!
+            ? lerpDouble(0.0, MediaQuery.paddingOf(context).top, curvedAnimation!.value)!
             : 0.0;
 
-          return FadeTransition(
-            opacity: CurvedAnimation(
+            viewFadeOnIntervalCurve ??= CurvedAnimation(
```
Comment (polina-c) [inline packages/flutter/lib/src/material/search_anchor.dart:643]: It seems there are indentation issues here.

**Comment by Dimilkalathiya** [comment]
@polina-c Thanks for review, fixed indentation issue and removed space

**Review by polina-c — APPROVED** [review]

**Pull request merged**